AlphaWallet/Core/BuyToken/BuyTokenService.swift

```diff
@@ -6,7 +6,51 @@
 //
 
 import Foundation
+import Combine
 
 protocol BuyTokenURLProviderType: TokenActionProvider {
-    func url(token: TokenActionsIdentifiable) -> URL?
+    func url(token: TokenActionsIdentifiable, wallet: Wallet) -> URL?
+}
+
+protocol TokenActionServiceProvidable {
+    func service(ofType: SupportedTokenActionsProvider.Type) -> SupportedTokenActionsProvider?
+}
+
+final class BuyTokenProvider: SupportedTokenActionsProvider, BuyTokenURLProviderType, TokenActionServiceProvidable {
+    private let subProviders: [BuyTokenURLProviderType & SupportedTokenActionsProvider]
+
+    var objectWillChange: AnyPublisher<Void, Never> {
+        return Just(subProviders)
+            .flatMap { Publishers.MergeMany($0.map { $0.objectWillChange }) }
+            .mapToVoid()
+            .eraseToAnyPublisher()
+    }
+    let analyticsNavigation: Analytics.Navigation = .fallback
+    let analyticsName: String = "Buy Crypto"
```

Comment:
Might be unused?

Reply:
yes, its looks unused but its required by protocol

Reply:
yes, its looks unused but its required by protocol